prov/efa/src/rxr/rxr_cq.c

```diff
@@ -475,27 +474,25 @@ void rxr_cq_complete_rx(struct rxr_ep *ep,
 	 * action of sending ctrl packet may cause the release of RX entry (when inject
 	 * was used on lower device).
 	 */
-	if (rx_entry->cq_entry.flags & FI_WRITE) {
+	if (op_entry->cq_entry.flags & FI_WRITE) {
 		/*
 		 * For write, only write RX completion when REMOTE_CQ_DATA is on
 		 */
-		if (rx_entry->cq_entry.flags & FI_REMOTE_CQ_DATA)
-			rxr_cq_write_rx_completion(ep, rx_entry);
-	} else if (rx_entry->cq_entry.flags & FI_READ) {
-		/* This rx_entry is part of the for emulated read protocol,
+		if (op_entry->cq_entry.flags & FI_REMOTE_CQ_DATA)
+			rxr_cq_write_rx_completion(ep, op_entry);
+	} else if (op_entry->cq_entry.flags & FI_READ) {
+		/* This op_entry is part of the for emulated read protocol,
```

Comment:
add an assertion here that `assert(op_entry->type == RXR_TX_ENTRY);`
also do `tx_entry = op_entry`, would help reader understand

Reply:
It is tx because rx_entry is removed on the requestor end. Data is received by the tx_entry on the requestor side.